src/language-js/printer-estree.js

```diff
@@ -4997,7 +4997,7 @@ function shouldHugType(node) {
         n.type === "VoidTypeAnnotation" ||
         n.type === "TSVoidKeyword" ||
         n.type === "NullLiteralTypeAnnotation" ||
-        n.type === "TSNullKeyword"
```

Comment:
`babel-ts` generates `TSNullKeyword` nodes. Also are we sure these changes are expected? I don't see anything about this in the changelog of `typescript-estree`.

Reply:
I didn't test locally, waiting for CI result.

This change didn't mentioned, but tests failed. https://github.com/prettier/prettier/runs/892776097#step:7:1263

#8806 also failed.

Reply:
You are right, `TSNullKeyword` can't be removed. Restored.

Reply:
@bradzacher It's not expected that the AST format changed for the `null` type literal (TS 3.9.7), is it?

Reply:
No, but I'm just now realising that I probably introduced a regression for <4.0.0 with the latest version. I'll submit a fix to undo it.

Reply:
Will it  change for typescript 4?

Reply:
Thank you!